tests/unit/test_client.py

```diff
@@ -138,3 +141,72 @@ async def test_CloudSQLClient_user_agent(
         assert client._user_agent == f"cloud-sql-python-connector/{version}+{driver}"
     # close client
     await client.close()
+
+
+async def test_cloud_sql_error_messages_get_metadata(
+    fake_credentials: Credentials,
+) -> None:
+    """
+    Test that Cloud SQL Admin API error messages are raised for _get_metadata.
+    """
+    # mock Cloud SQL Admin API calls with exceptions
+    client = CloudSQLClient(
+        sqladmin_api_endpoint="https://sqladmin.googleapis.com",
+        quota_project=None,
+        credentials=fake_credentials,
+    )
+    get_url = "https://sqladmin.googleapis.com/sql/v1beta4/projects/my-project/instances/my-instance/connectSettings"
+    resp_body = {
+        "error": {
+            "code": 403,
+            "message": "Cloud SQL Admin API has not been used in project 123456789 before or it is disabled",
+        }
+    }
+    with aioresponses() as mocked:
+        mocked.get(
+            get_url,
+            status=403,
+            payload=resp_body,
+            repeat=True,
+        )
+        with pytest.raises(ClientResponseError) as exc_info:
+            await client._get_metadata("my-project", "my-region", "my-instance")
+            assert exc_info.status == 403
```

Comment:
This needs to be `exc_info.value.status == 403`. Similar below: it should be `exc_info.value.message == "..."`.

`exc_info` is an `ExceptionInfo` object. To get the actual `ClientResponseError` object from it, we get it from `exc_info.value`. More info: 
![8VjETmTRMP2hDfP](https://github.com/user-attachments/assets/f7592165-3876-4555-aee5-b7c402923afd)

Reply:
Another good catch @rhatgadkar-goog 👏 

That matches what I have in another test

https://github.com/GoogleCloudPlatform/cloud-sql-python-connector/blob/968b6b2b5691d00edd164fdc09a3054faccbc2c6/tests/unit/test_connector.py#L88-L95